modules/utils.lua

```diff
@@ -142,6 +142,21 @@ function GenerateRandomPlate(pattern) -- luacheck: ignore
     return newPattern:upper()
 end
 
+--- Returns a mapped table of vehicles by category
+--- @param Vehicles table
+--- @return table
+function GetVehiclesByCategory(Vehicles)
+	local mappedVehicles = {}
+	for vehicleIndex, vehicleData in pairs(Vehicles) do
+		if not mappedVehicles[vehicleData.category] then
+			mappedVehicles[vehicleData.category] = {}
+		end
+
+		mappedVehicles[vehicleData.category][vehicleIndex] = vehicleData
+	end
+	return mappedVehicles
+end
+
```

Comment:
This should not be a module function. Move this to where GetVehiclesByHash is calculated as this is similar; a transform of the original data table. Consistency is important, if we're going to do transforms as module functions we should do them for vehicles by hash as well. A good general util function might be to create a mapping of any common subfield for any table..

Reply:
better now? :)